### DISCUSSION TITLE
[Vim9script] There is a reason why def function must be in pascal cas/with the first letter a capital?

### DISCUSSION THREAD

**Discussion by saccarosium** [vim/vim#14873] [Q&A]
I was just wondering why this decision was made. There was a particular problem that it was trying to solve, or it was to make a common style pattern.

**Comment by chrisbra** [comment]
I believe the main reason was, so that it cannot shadow builtin functions

**Comment by thingsiplay** [comment]
Functions defined with `def` keyword has some differences to old style functions. So maybe this is a way to make that clear to the caller. I'm glancing over the rationale section in the documentation for functions and looks like chrisbra was on track:

> Since [a](https://vimhelp.org/insert.txt.html#a) [script-local](https://vimhelp.org/map.txt.html#script-local) function [reference](https://vimhelp.org/intro.txt.html#reference) can be used without "s:" the name [must](https://vimhelp.org/vim_faq.txt.html#must)
> start with an upper [case](https://vimhelp.org/change.txt.html#case) [letter](https://vimhelp.org/print.txt.html#letter) even when using the "s:" prefix.  In legacy
> [script](https://vimhelp.org/usr_41.txt.html#script) "s:funcref" could be used, because [it](https://vimhelp.org/motion.txt.html#it) could not be referred to with
> "funcref"[.](https://vimhelp.org/repeat.txt.html#.)  In [Vim9](https://vimhelp.org/vim9.txt.html#Vim9) [script](https://vimhelp.org/usr_41.txt.html#script) [it](https://vimhelp.org/motion.txt.html#it) can, therefore "s:Funcref" [must](https://vimhelp.org/vim_faq.txt.html#must) be used to avoid
> that the name interferes with builtin functions.

\-- https://vimhelp.org/vim9.txt.html#vim9-rationale

BTW this question is marked as an announcement and popped up in my feed.

**Comment by N-R-K** [comment]
> BTW this question is marked as an announcement and popped up in my feed.

Same. I believe github defaults to "announcement" when a maintainer tries to open a discussion. Which is a poor default since "announcements" are meant to be low-volume important notices.
